include/astshim/FrameDict.h

```diff
@@ -105,8 +105,10 @@ class FrameDict : public FrameSet {
 
     @throws std::invalid_argument if two Frames in the FrameSet have the same non-empty domain.
     */
-    explicit FrameDict(FrameSet const &frameSet)
-            : FrameDict(reinterpret_cast<AstFrameSet *>(frameSet.copy()->getRawPtr())) {}
+    explicit FrameDict(FrameSet const &frameSet) : FrameSet(std::move(*frameSet.copy())), _domainIndexDict() {
+        _domainIndexDict = _makeNewDict(*this);
+    }
+
```

Comment:
Is the constructor from an `AstFrameSet*` still used for anything?

Reply:
Yes, it is used by Object::makeShim